src/bun.js/bindings/webcore/JSDOMFormData.cpp

```diff
@@ -462,19 +462,19 @@ static inline JSC::EncodedJSValue jsDOMFormDataPrototypeFunction_set2Body(JSC::J
     RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
     EnsureStillAliveScope argument1 = callFrame->uncheckedArgument(1);
 
-    EnsureStillAliveScope argument2 = callFrame->argument(2);
-    auto filename = argument2.value().isUndefined() ? String() : convert<IDLUSVString>(*lexicalGlobalObject, argument2.value());
-    RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
-
     RefPtr<Blob> blobValue = nullptr;
     if (argument1.value().inherits<JSBlob>()) {
         blobValue = Blob::create(argument1.value());
     }
 
     if (!blobValue) {
         throwTypeError(lexicalGlobalObject, throwScope, "Expected argument to be a Blob."_s);
-        RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
     }
+    RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
+
+    EnsureStillAliveScope argument2 = callFrame->argument(2);
```

Comment:
We should keep this in the original spot it was in because in certain cases the callFrame can be moved which leads to the GC not realizing the value still needs to be kept alive

Reply:
ah cool. i wasn't sure about that. will update.